specs/opcodes/0x80_dupx.md

```diff
@@ -0,0 +1,21 @@
+# DUPX op code
+## Procedure
+   dupx represents op codes of dup1....dup16. which picks up value at 'x' position inside the stack, then push the value to stack. 
+
+## Constraints
+   1. opId = OpcodeId(0x80...0x16)
+   2. state transition:  
+      gc + 2 (read + write)
+      stack_pointer - 1
+      pc + 1
+      gas + 3
+   3. lookups: 1 range lookup + 2 bussmapping lookups:
+      position 'x' range from [1..16]
+      operand must come from position 'x' inside stack
+      position 'x' value equals stack top value when this operation done
+
+## Exceptions
+   1. gas out: remaining gas is not enough 
+   2. stack overflow: when stack is full, which means stack pointer is 0 before dupx
```

Comment:
Do you think we need an underflow check? Let's say dup 9, if your stack has less than 9 items then it's an underflow.

References in Go-ethereum

https://github.com/ethereum/go-ethereum/blob/94451c2788295901c302c9bf5fa2f7b021c924e2/core/vm/jump_table.go#L807

https://github.com/ethereum/go-ethereum/blob/1d995731923ca899964371ddb213d40b7e773818/core/vm/interpreter.go#L202-L205

Reply:
Agree with @ChihChengLiang, I think underflow can occur. To empirically check this we can run https://github.com/ethereum/go-ethereum/blob/master/tests/state_test.go#L29 which will also run tests in tests/testdata/GeneralStateTests/stStackTests, and we set into https://github.com/ethereum/go-ethereum/blob/master/core/vm/interpreter.go#L203 something like:
if strings.Contains(op.String(), "DUP") {
    fmt.Println("DUP underflow")
}